Adding Soroban to Stellar Fundamentals (#269)
stellar/stellar-docs: docs/fundamentals-and-concepts/networks.mdx

---
title: Networks
sidebar_position: 30
---

Stellar has three networks: the public network (Mainnet, also called Pubnet or the Public Network), the test network (Testnet), and a dev network (Futurenet). Mainnet is the main network used by applications in production. It connects to real financial rails and requires XLM to cover minimum balances, transaction fees, and rent. The Testnet is a smaller, free-to-use network maintained by SDF that functions like the Mainnet but doesn’t connect to real money. It has a built-in testnet XLM faucet (called Friendbot), and it's the best place for developers to test their applications. Futurenet is a dev network you can use to test more bleeding edge features that also has access to its own Friendbot.

## Stats: Mainnet versus Testnet versus Futurenet

### Mainnet

- Validator nodes are run by the public
- SDF offers a free [Horizon instance](https://horizon.stellar.org/) to interact with the Mainnet with a limited set of history, or you can [run your own](https://developers.stellar.org/docs/run-api-server)
- You need to fund your account with XLM from another account

> - Mainnet is limited to 1,000 operations per ledger and will be limited to a maximum of 30 smart contract transactions per ledger (the precise amount of smart contract txs per ledger can vary greatly depending on transaction [resource limits](https://soroban.stellar.org/docs/fundamentals-and-concepts/fees-and-metering#resource-limits))

Reviewer: ```suggestion
- Mainnet is limited to 1,000 operations per ledger and will be limited to somewhere around 30 smart contract transactions per ledger when the phased rollout of Protocol 20 is complete in Q1 2024 (the precise amount of smart contract txs per ledger can vary greatly depending on transaction [resource limits](https://soroban.stellar.org/docs/fundamentals-and-concepts/fees-and-metering#resource-limits))
```

  - See more detailed smart contract network settings in the section on [Fees and Metering](https://soroban.stellar.org/docs/fundamentals-and-concepts/fees-and-metering) in the Soroban docs
- No publicly available RPC, see RPC service providers [here](https://soroban.stellar.org/docs/reference/rpc-list)

### Testnet

- SDF runs three core validator nodes
- SDF offers a free [Horizon instance](https://horizon-testnet.stellar.org/) you can use to interact with the Testnet
- Friendbot is a faucet you can use for free Testnet XLM

> - Testnet is limited to 100 operations per ledger and one smart contract transaction per ledger

Reviewer: Is one smart contract transaction per ledger accurate?  Just want to verify that's the case before we enshrine it in the docs.

- SDF offers free RPC endpoints, more information [here](https://soroban.stellar.org/docs/reference/rpc-list#sdf-futurenet-and-testnet-only)

## Friendbot

Friendbot is a bot that funds accounts with fake XLM on Testnet or Futurenet. You can request XLM from Friendbot using the Stellar Laboratory or with various SDKs. Requests to Friendbot are rate limited, so use it wisely. Friendbot provides 10,000 fake XLM when funding a new account.

If you are creating multiple accounts, you can fund your first account with Friendbot and then use that first account to fund your subsequent accounts using the Create Account operation.

## Testnet and Futurenet data reset

Testnet and Futurenet are reset periodically to the genesis ledger to declutter the network, remove spam, reduce the time needed to catch up on the latest ledger, and help maintain the system. Resets clear all ledger entries (accounts, trustlines, offers, smart contract data, etc.), transactions, and historical data from Stellar Core, Horizon, and the Soroban RPC- which is why developers should not rely on the persistence of accounts or the state of any balances when using Testnet or Futurenet.

> Futurenet resets are on a less regular cadence than Testnet resets and don't have a set schedule.

Reviewer: ```suggestion
Futurenet resets happen when necessary, and often with very little warning. Testnet resets happen once per quarter on a set schedule.
```

> Testnet resets happen once per quarter at 17:00 UTC and are announced at least two weeks in advance on the [Stellar Dashboard](http://dashboard.stellar.org/) and through several developer community channels. Here are the 2023 dates:

Reviewer: ```suggestion
Testnet resets happen once per quarter at 17:00 UTC and are announced at least two weeks in advance on the [Stellar Dashboard](http://dashboard.stellar.org/) and through several developer community channels. Here are the 2024 dates:
```

> March 15, 2023  

Reviewer: ```suggestion
March 13, 2024  
```

> June 14, 2023  

Reviewer: ```suggestion
June 12, 2024  
```

> ~~September 13, 2023~~ Note: this reset was skipped due to its proximity to the Protocol 20 Testnet upgrade.  

Reviewer: ```suggestion
September 11, 2024 
```

> December 18, 2023

Reviewer: ```suggestion
December 11, 2024
```

---
title: Networks
sidebar_position: 30
---

> Stellar has three networks: the public network (Mainnet, also called Pubnet or the Public Network), the test network (Testnet), and a dev network (Futurenet). Mainnet is the main network used by applications in production. It connects to real rails and requires XLM to cover minimum balances, transaction fees, and rent. The Testnet is a smaller, free-to-use network maintained by SDF that functions like the Mainnet but doesn’t connect to real money. It has a built-in testnet XLM faucet (called Friendbot), and it's the best place for developers to test their applications. Futurenet is a dev network you can use to test more bleeding edge features that also has access to Friendbot.

Reviewer 2: ```suggestion
Stellar has three networks: the public network (Mainnet, also called Pubnet or the Public Network), the test network (Testnet), and a dev network (Futurenet). Mainnet is the main network used by applications in production. It connects to real financial rails and requires XLM to cover minimum balances, transaction fees, and rent. The Testnet is a smaller, free-to-use network maintained by SDF that functions like the Mainnet but doesn’t connect to real money. It has a built-in testnet XLM faucet (called Friendbot), and it's the best place for developers to test their applications. Futurenet is a dev network you can use to test more bleeding edge features that also has access to its own Friendbot.
```

## Stats: Mainnet versus Testnet versus Futurenet

### Mainnet

- Validator nodes are run by the public

> - SDF offers a free [Horizon instance](https://horizon.stellar.org/) to interact with the Pubnet, or you can run your own

Reviewer 2: ```suggestion
- SDF offers a free [Horizon instance](https://horizon.stellar.org/) to interact with the Pubnet with a limited set of history, or you can [run your own](https://developers.stellar.org/docs/run-api-server)
```

  - See more detailed smart contract network settings in the section on [Fees and Metering](https://soroban.stellar.org/docs/fundamentals-and-concepts/fees-and-metering) in the Soroban docs
- No publicly available RPC, see RPC service providers [here](https://soroban.stellar.org/docs/reference/rpc-list)

### Testnet

- SDF runs three core validator nodes
- SDF offers a free [Horizon instance](https://horizon-testnet.stellar.org/) you can use to interact with the Testnet
- Friendbot is a faucet you can use for free Testnet XLM
- Testnet is limited to 100 operations per ledger
- SDF offers free RPC endpoints, more information [here](https://soroban.stellar.org/docs/reference/rpc-list#sdf-futurenet-and-testnet-only)

### Futurenet

- SDF runs core validator nodes
- SDF offers a free [Horizon instance](https://horizon-futurenet.stellar.org) you can use to interact with the Futurenet
- Friendbot is a faucet you can use for free Futurenet XLM

> - _What are Futurenet's transaction limitations?_

Author: @ElliotFriend

A script can automate this entire process by creating an account with friendbot and submitting a set of transactions that are predefined as a part of your testing infrastructure.

## What Testnet and Futurenetshould and should not be used for

### Testnet is good for

- Creating test accounts (with funding from Friendbot);
- Developing applications and contracts and exploring tutorials on Stellar without the potential to lose any assets;
- Testing existing applications against new releases or release candidates of Stellar Core, Horizon, and the Soroban RPC;
- Performing data analysis on a smaller, non-trivial data set compared to the Mainnet.

### Testnet is bad for

- Load and stress testing;
- High availability test infrastructure- SDF does not guarantee Testnet availability;
- Long-term storage of data on the network since the network resets periodically;
- A testing infrastructure that requires more control over the test environment, such as:

  - The ability to control the data reset frequency;
  - The need to secure private or sensitive data (before launching on the Mainnet). You can always run your own test network for use cases that don’t work well with SDF’s Testnet.

>   ### Futurenet is good for

Author: @ElliotFriend let's discuss
